crates/task-impls/src/da.rs

```diff
@@ -108,35 +108,31 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> DaTaskState<TYP
                 // the `DaProposalRecv` event. Otherwise, the view number subtraction below will
                 // cause an overflow error.
                 // TODO ED Come back to this - we probably don't need this, but we should also never receive a DAC where this fails, investigate block ready so it doesn't make one for the genesis block
-                if self.cur_view != TYPES::Time::genesis() && view < self.cur_view - 1 {
-                    warn!("Throwing away DA proposal that is more than one view older");
-                    return None;
-                }
+                ensure!(
+                    self.cur_view == TYPES::Time::genesis() || view >= self.cur_view - 1,
+                    "Throwing away DA proposal that is more than one view older"
+                );
 
-                if self
+                ensure!(!self
                     .consensus
                     .read()
                     .await
                     .saved_payloads()
-                    .contains_key(&view)
-                {
-                    warn!("Received DA proposal for view {:?} but we already have a payload for that view.  Throwing it away", view);
-                    return None;
-                }
+                    .contains_key(&view),
+                    format!("Received DA proposal for view {:?} but we already have a payload for that view.  Throwing it away", view));
 
                 let encoded_transactions_hash = Sha256::digest(&proposal.data.encoded_transactions);
 
                 // ED Is this the right leader?
-                let view_leader_key = self.da_membership.leader(view);
-                if view_leader_key != sender {
-                    error!("DA proposal doesn't have expected leader key for view {} \n DA proposal is: {:?}", *view, proposal.data.clone());
-                    return None;
-                }
+                let view_leader_key = self.da_membership.leader(view)?;
+                ensure!(view_leader_key == sender,
+                    format!("DA proposal doesn't have expected leader key for view {} \n DA proposal is: {:?}", *view, proposal.data.clone())
+                );
 
-                if !view_leader_key.validate(&proposal.signature, &encoded_transactions_hash) {
-                    error!("Could not verify proposal.");
-                    return None;
-                }
+                ensure!(
+                    view_leader_key.validate(&proposal.signature, &encoded_transactions_hash),
+                    "Could not verify proposal."
+                );
```

Comment:
This is downgrading the error to info log level.  I think this one is at least a warn level event (someone is sending us malformed proposals)

Reply:
Set it to warn!